proto/destination.proto

```diff
@@ -141,6 +149,30 @@ message Route {
   repeated ResponseClass response_classes = 2;
   // Metric labels to attach to requests and responses that match this route.
   map<string, string> metrics_labels = 3;
+  // If a route is retryable, any failed requests on that route may be retried
+  // by the proxy.
+  bool is_retryable = 4;
+  // After this time has elapsed since receiving the initial request, any
+  // outstanding request will be cancelled, a timeout error response will be
+  // returned, and no more retries will be attempted.
+  google.protobuf.Duration timeout = 5;
```

Comment:
Open question: should timeout be measured from when the initial request headers are received to when the final response is complete?  i.e. if the response headers have been received but the response body is still streaming when the timeout elapses, should the response stream be reset?